pytorch_lightning/callbacks/finetuning.py

```diff
@@ -0,0 +1,221 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Finetunning Callback
+^^^^^^^^^^^^^^^^^^^^
+Freeze and unfreeze models for finetunning purposes
+"""
+from typing import Callable, Generator, Optional
+
+import torch
+from torch.nn import Module
+from torch.nn.modules.container import Sequential
+from torch.optim.optimizer import Optimizer
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+BN_TYPES = (torch.nn.BatchNorm1d, torch.nn.BatchNorm2d, torch.nn.BatchNorm3d)
+
+
+def _make_trainable(module: Module) -> None:
```

Comment:
Why don't you put these functions inside of the Callback? Users might want to override them

Reply:
Done !